### PR TITLE
(kubernetes) fix issues discovered upgrading to 1.4.

### PR DESCRIPTION
**Pull request by lwander**
This includes a change in the label semantics, behavior of IntOrString, as well as the annotations keymap.

@duftler PTAL, @gardleopard FYI

### CODE REVIEW

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/deploy/ops/servergroup/DestroyKubernetesAtomicOperation.groovy [hunk 1]
```diff
@@ -63,6 +63,8 @@ class DestroyKubernetesAtomicOperation implements AtomicOperation<Void> {
     if (deployment && replicaSet) {
       task.updateStatus BASE_PHASE, "Checking if deployment ${deploymentName} needs to be destroyed..."
       // If we selected to delete the replica set in the currently active deployment, this will delete everything owned by the deployment.
+      print ",, d ${credentials.apiAdaptor.getDeploymentRevision(deployment)}"
```
Comment (duftler) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/deploy/ops/servergroup/DestroyKubernetesAtomicOperation.groovy]: Extraneous `print` statements.

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/deploy/ops/servergroup/DeployKubernetesAtomicOperation.groovy [hunk 2]
```diff
@@ -101,11 +101,6 @@ class DeployKubernetesAtomicOperation implements AtomicOperation<DeploymentResul
       }
     }
 
-
-    task.updateStatus BASE_PHASE, "Sending replica set spec to the Kubernetes master."
```
Comment (duftler) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/deploy/ops/servergroup/DeployKubernetesAtomicOperation.groovy]: Why were these removed? Did some other log msg take their place?

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesApiConverter.groovy [hunk 3]
```diff
@@ -808,11 +808,19 @@ class KubernetesApiConverter {
         builder = builder.withNewRollingUpdate()
 
         if (rollingUpdate.maxSurge) {
-          builder = builder.withNewMaxSurge(rollingUpdate.maxSurge)
+          def maxSurge = rollingUpdate.maxSurge
```
Comment (duftler) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesApiConverter.groovy]: I didn't chase down the api docs, but I'm guessing int is preferred, but either int or string is ok?
Maybe add a comment explaining why the extra complication here, including what exception types are possible (guessing NPE and NFE).

Or you could pull this logic out into a method that returns an `IntOrString` and use it here and below.

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/deploy/ops/servergroup/DeployKubernetesAtomicOperation.groovy (hunk 2 above)
Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/deploy/ops/servergroup/DeployKubernetesAtomicOperation.groovy]: No I should move them

File: clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesApiConverter.groovy (hunk 3 above)
Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesApiConverter.groovy]: I'd prefer to pull these into separate methods, but the builder is overloaded to take an int or a string, and the builder changes type with each invocation depending on how nested you are in the build so you can't pass that to a helper.

I'll clean up the exception handler for clarity

Comment (duftler) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesApiConverter.groovy]: Makes sense.
There is apparently also a groovy `isInteger()` method you can use to avoid the try/catch technique entirely.

Comment (lwander) [inline clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/api/KubernetesApiConverter.groovy]: Ah much nicer
